### PR TITLE
En 5893 rating config refactor and add meta rating steps

### PR DESCRIPTION
**Pull request by raduchis**
refactored economics config and extracted another file ratings config;
added different ratingSteps for shards and meta

### CODE REVIEW

File: cmd/node/config/economics.toml [hunk 1]
```diff
@@ -27,46 +27,4 @@
...
-    [[RatingSettings.SelectionChance]]
-    MaxThreshold = 800000
-    ChancePercent = 22
-    [[RatingSettings.SelectionChance]]
-    MaxThreshold = 900000
-    ChancePercent = 23
-    [[RatingSettings.SelectionChance]]
-    MaxThreshold = 1000000
-    ChancePercent = 24
+    UnJailValue = "1000"
```
Comment (iulianpascalau) [inline cmd/node/config/economics.toml]: Please add an empty line here

File: cmd/node/config/ratings.toml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#Ratings configuration
+
+[General]
+    StartRating = 500000
+    MaxRating = 1000000
+    MinRating = 1
+    SelectionChances = [
+        { MaxThreshold = 0, ChancePercent = 5},
```
Comment (iulianpascalau) [inline cmd/node/config/ratings.toml]: :+1:

File: cmd/node/config/ratings.toml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        { MaxThreshold = 500000,ChancePercent = 19},
+        { MaxThreshold = 600000,ChancePercent = 20},
+        { MaxThreshold = 700000,ChancePercent = 21},
+        { MaxThreshold = 800000,ChancePercent = 22},
+        { MaxThreshold = 900000,ChancePercent = 23},
+        { MaxThreshold = 1000000,ChancePercent = 24},
+    ]
+
+[ShardChain.RatingSteps]
+    ProposerIncreaseRatingStep = 405
```
Comment (iulianpascalau) [inline cmd/node/config/ratings.toml]: Maybe add some comments here (containing succint formulas) as to be able to compute the values on different testnet configurations

File: process/rating/ratingsData.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	startRating          uint32
+	maxRating            uint32
+	minRating            uint32
+	metaRatingsStepData  process.RatingsStepHandler
+	shardRatingsStepData process.RatingsStepHandler
+	selectionChances     []process.SelectionChance
+}
+
+// NewRatingsData creates a new RatingsData instance
+func NewRatingsData(settings *config.RatingsConfig) (*RatingsData, error) {
```
Comment (iulianpascalau) [inline process/rating/ratingsData.go]: Configs can be passed as value. One nil check (settings == nil) will no longer be required.

File: process/rating/ratingsData.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+	if settings.General.MinRating < 1 {
+		return nil, process.ErrMinRatingSmallerThanOne
+	}
+	if settings.General.MinRating > settings.General.MaxRating {
+		return nil, process.ErrMaxRatingIsSmallerThanMinRating
```
Comment (iulianpascalau) [inline process/rating/ratingsData.go]: might wrap this error in order to read the minimum/provided values

File: process/rating/ratingsData.go [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+	}
+	if settings.General.MaxRating < settings.General.StartRating || settings.General.MinRating > settings.General.StartRating {
+		return nil, process.ErrStartRatingNotBetweenMinAndMax
```
Comment (iulianpascalau) [inline process/rating/ratingsData.go]: might wrap this error in order to read the minimum/provided values

File: process/rating/ratingStepData.go [hunk 7]
```diff
@@ -0,0 +1,44 @@
+package rating
+
+import (
```
Comment (iulianpascalau) [inline process/rating/ratingStepData.go]: Maybe add one unit test for this structure in order to validate the getters?

File: process/peer/process_test.go [hunk 8]
```diff
@@ -1246,9 +1237,9 @@ func TestValidatorStatisticsProcessor_CheckForMissedBlocksWithRoundDifferences(t
 				want: result{
 					decreaseValidatorValue: intValidatorProbability,
 					decreaseLeaderValue:    intLeaderProbability,
-					tempRating: rater.StartRating -
-						intLeaderProbability*rater.DecreaseProposer -
-						intValidatorProbability*rater.DecreaseValidator,
+					tempRating: uint32(int32(rater.StartRating) +
```
Comment (AdoAdoAdo) [inline process/peer/process_test.go]: is it possible to overflow? 
e.g uint32(negative value)

File: sharding/indexHashedNodesCoordinator.go [hunk 9]
```diff
@@ -681,28 +681,41 @@ func (ihgs *indexHashedNodesCoordinator) IsInterfaceNil() bool {
...
-		for _, v := range validators {
+		for _, v := range waiting[shardId] {
 			pk := v.PubKey()
 			log.Debug("waiting", "pk", pk, "shardId", shardId)
 		}
 	}
 
-	for _, v := range leaving {
-		pk := v.PubKey()
+	for _, validator := range leaving {
```
Comment (AdoAdoAdo) [inline sharding/indexHashedNodesCoordinator.go]: "validator" may shadow the validator type in the same package, so can you use different variable name?

File: sharding/indexHashedNodesCoordinator.go [hunk 10]
```diff
@@ -681,28 +681,41 @@ func (ihgs *indexHashedNodesCoordinator) IsInterfaceNil() bool {
...
+		pk := validator.PubKey()
+		log.Debug("eligible", "pk", pk, "shardId", core.MetachainShardId)
+	}
+	for _, validator := range waiting[core.MetachainShardId] {
+		pk := validator.PubKey()
+		log.Debug("waiting", "pk", pk, "shardId", core.MetachainShardId)
+	}
+
+	for shardId := uint32(0); shardId < nbShards; shardId++ {
+		for _, v := range eligible[shardId] {
```
Comment (AdoAdoAdo) [inline sharding/indexHashedNodesCoordinator.go]: you change this for to 
```go
for shard := uint32(0); shard <= nbShards; shard++ {
  shardID := shard 
  if shardID == nbShards{
    shardID = core.MetachainShardId
  }
...
}
```
then you could also remove previous two for loops for the metachain

File: cmd/node/config/economics.toml (hunk 1 above)
Comment (raduchis) [reply cmd/node/config/economics.toml]: fixed

File: cmd/node/config/ratings.toml (hunk 3 above)
Comment (raduchis) [reply cmd/node/config/ratings.toml]: added to the begining of rating.toml

File: process/rating/ratingsData.go (hunk 5 above)
Comment (raduchis) [reply process/rating/ratingsData.go]: fixed

File: process/rating/ratingsData.go (hunk 6 above)
Comment (raduchis) [reply process/rating/ratingsData.go]: fixed

File: process/rating/ratingStepData.go (hunk 7 above)
Comment (raduchis) [reply process/rating/ratingStepData.go]: fixed

File: config/ratingsConfig.go [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+	RatingSteps
+}
+
+//RatingValue will hold different rating options with increase and decrease steps
+type RatingValue struct {
+	Name  string
+	Value int32
+}
+
+//RatingValue will hold different rating options with increase and decresea steps
```
[the remark this answers is not in the document]
Comment (raduchis) [reply config/ratingsConfig.go]: fixed

File: sharding/indexHashedNodesCoordinator.go (hunk 10 above)
Comment (raduchis) [reply sharding/indexHashedNodesCoordinator.go]: fixed

File: sharding/indexHashedNodesCoordinator.go (hunk 9 above)
Comment (raduchis) [reply sharding/indexHashedNodesCoordinator.go]: fixed

File: process/peer/process_test.go (hunk 8 above)
Comment (raduchis) [reply process/peer/process_test.go]: it is just for testing purposes and with the current configuration it shouldn't happen

File: process/rating/ratingsData.go (hunk 4 above)
Comment (raduchis) [reply process/rating/ratingsData.go]: fixed

File: integrationTests/mock/raterMock.go [hunk 12]
```diff
@@ -5,14 +5,13 @@ import "github.com/ElrondNetwork/elrond-go/sharding"
 // RaterMock -
 type RaterMock struct {
 	GetRatingCalled                  func(string) uint32
```
Comment (iulianpascalau) [inline integrationTests/mock/raterMock.go]: indentation fix

**Pull request merged**